Documentation/op-guide/configuration.md

```diff
@@ -270,6 +270,11 @@ The security flags help to [build a secure etcd cluster][security].
 + default: ""
 + env variable: ETCD_CLIENT_CRL_FILE
 
+### --client-cert-allowed-name
```

Comment:
allowed-san?

Reply:
When checking whether a given hostname is valid for a TLS certificate, it's typical to not mandate which particular field the name comes from. The [implementation of `(*x509.Certificate).VerifyHostname()`](https://github.com/golang/go/blob/go1.12.6/src/crypto/x509/verify.go#L994-L1028) uses SAN if any are set, and falls back to CN.

Reply:
OK. The implementation checks CN first, then checks parsed SAN, right? Can we make this clear?

Reply:
It is hard to infer this behavior from the flag and current help message.

Reply:
I believe the verification logic is something like this:
```
func checkName(name string) bool {
  if len(SAN) == 0 {
    return validHostname(CN) && matches(CN, name)
  }
  for _, sanEntry := range SAN {
    if matches(sanEntry, name) {
      return true
    }
  }
  return false
}
```

The full behavior is described by [RFC 6125 section 6.4](https://tools.ietf.org/html/rfc6125#section-6.4), which is too long to describe here.